sources/packages/web/src/components/common/IdleTimeChecker.vue

```diff
@@ -0,0 +1,104 @@
+<template>
+  <div
+    class="v-main body-background"
+    @mouseover="setLastActivityTime"
+    @click="setLastActivityTime"
+    @keyup="setLastActivityTime"
+  >
+    <slot></slot>
+    <ConfirmExtendTime ref="extendTimeModal" :clientIdType="clientIdType" />
+  </div>
+</template>
+
+<script lang="ts">
+import { onMounted, ref, onUnmounted, computed } from "vue";
+import { ClientIdType } from "@/types";
+import { ModalDialog, useInstitutionAuth, useFormatters } from "@/composables";
+import {
+  MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER,
+  MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT,
+  MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION,
+  MINIMUM_IDLE_TIME_FOR_WARNING_AEST,
+} from "@/constants/system-constants";
+import ConfirmExtendTime from "@/components/common/modals/ConfirmExtendTime.vue";
+
+export default {
+  components: { ConfirmExtendTime },
+  props: {
+    clientIdType: {
+      type: String,
+      required: true,
+      default: "" as ClientIdType,
+    },
+  },
+  setup(props: any) {
+    const lastActivityLogin = ref(new Date());
+    const interval = ref();
+    const extendTimeModal = ref({} as ModalDialog<boolean>);
+    const { isAuthenticated } = useInstitutionAuth();
+    const { getDatesDiff } = useFormatters();
+
+    const minimumIdleTime = computed(() => {
+      switch (props.clientIdType) {
+        case ClientIdType.Student:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_STUDENT;
+        case ClientIdType.Institution:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_INSTITUTION;
+        case ClientIdType.SupportingUsers:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_SUPPORTING_USER;
+        case ClientIdType.AEST:
+          return MINIMUM_IDLE_TIME_FOR_WARNING_AEST;
+        default:
+          return 0.02;
+      }
+    });
+
+    const startIdleCheckerTimer = () => {
+      if (isAuthenticated.value) {
```

Comment:
Whatever starting the timer, would not be good to reset the timer?
```
lastActivityLogin.value = new Date();
```

Reply:
updated